.travis.yml

```diff
@@ -31,11 +31,23 @@ addons:
     - conntrack
 script: |
   "${MAKE}" unit build
-  if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
-    sudo minikube start # vm-driver=none requires root
-    sudo chown -R "$USER" "$HOME/.kube" "$HOME/.minikube"
-    KUBECONFIG="$HOME/.kube/config" make e2e
+  if [[ "$TRAVIS_OS_NAME" != "linux" ]]; then
+    echo "skipping e2e tests for ${TRAVIS_OS_NAME}"
+    exit 0
   fi
+
+  if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then
+    # no Quay.io credentials available for PR tests
+    echo "starting local test docker registry"
+    sudo docker run -d -p 5000:5000 --restart=always --name registry registry:2
+    export DOCKER_REGISTRY_HOST="localhost:5000"
+    printf "[registries.insecure]\nregistries = ['localhost:5000']\n" > /etc/containers/registries.conf
```

Comment:
I ran into this issue when testing image polling on OLM, but there the openshift CI included the openshift registry so I was able to use that. I'm wondering if generating some certs on a per-CI run basis and passing those to the registry would make more sense than running everything with `skipTLScheck` since in production we would expect users to be hosting their images on a registry that has certs and our tests end up closer to the real thing